personalized_nlp/learning/train.py

```diff
@@ -69,14 +69,14 @@ def train_test(
     )
     progressbar_checkpoint = TQDMProgressBar(refresh_rate=20)
 
-    _use_cuda = use_cuda and torch.cuda.is_available()
+    # _use_cuda = use_cuda and torch.cuda.is_available()
 
     callbacks = [checkpoint_callback, progressbar_checkpoint]
     if custom_callbacks is not None:
         callbacks = callbacks + custom_callbacks
 
     trainer = pl.Trainer(
-        gpus=1 if _use_cuda else 0,
+        # gpus=1 if _use_cuda else 0,
```

Comment:
Da się bez tego sterować tym czy ma być użyte CPU czy GPU?

Reply:
Zgodnie z dokumentacją nowszych wersji PyTorch Lightning

https://lightning.ai/docs/pytorch/stable/common/trainer.html#trainer-class-api

należy używać parametrów `devices` oraz `accelerator` zamiast parametru `gpus`, który nie jest już wspierany.

Przykładowo:

`Trainer(gpus=4) -> Trainer(devices=4, accelerator='gpu')`
`Trainer(gpus=[1,2,3]) -> Trainer(devices=[1,2,3], accelerator='gpu')`

Z mojego doświadczenia zauważyłem, że zachowanie Trainera jest tutaj powiązane z innymi miejscami w kodzie, gdzie używane jest `CUDA_VISIBLE_DEVICES`, ale mogę dodać tutaj w Trainerze wprost te dwa parametry z najnowszej wersji PyTorch Lightning.

Reply:
Z tego co widzę to nowa wersja PL faktycznie powinna używać GPU jeśli jest dostępne, a parametr `accelerator` jest ustawiony na `auto`, więc można to tak zostawić :)